.github/workflows/publish_snapshot.yml

```diff
@@ -0,0 +1,95 @@
+name: Publish snapshots to maven
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build-and-publish-snapshots:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: write
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK 8
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '8'
+          cache: 'gradle'
+
+      - name: 'Setup: Java 8 env'
+        run: echo "JAVA8_HOME=$JAVA_HOME" >> $GITHUB_ENV
+
+      - name: Set up JDK 11
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '11'
+          cache: 'gradle'
+
+      - name: 'Setup: Java 11 env'
+        run: echo "JAVA11_HOME=$JAVA_HOME" >> $GITHUB_ENV
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+          cache: 'gradle'
+
+      - name: 'Setup: Java 17 env'
+        run: echo "JAVA17_HOME=$JAVA_HOME" >> $GITHUB_ENV
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.PUBLISH_SNAPSHOTS_ROLE }}
+          aws-region: us-east-1
+
+      - name: Read value from Properties-file
+        id: read_property
+        uses: christian-draeger/read-properties@1.1.1
+        with:
+          path: 'buildSrc/opensearch-hadoop-version.properties'
+          properties: 'opensearch_hadoop'
+
+      - name: publish snapshots to build repositories
+        run: |
+          export SONATYPE_USERNAME=$(aws secretsmanager get-secret-value --secret-id maven-snapshots-username --query SecretString --output text)
+          export SONATYPE_PASSWORD=$(aws secretsmanager get-secret-value --secret-id maven-snapshots-password --query SecretString --output text)
+          echo "::add-mask::$SONATYPE_USERNAME"
+          echo "::add-mask::$SONATYPE_PASSWORD"
+          ./gradlew :opensearch-hadoop-mr:publishMainPublicationToBuildRepository
+          ./gradlew :opensearch-hadoop-hive:publishMainPublicationToBuildRepository
+          ./gradlew -S -Dbuild.snapshot=true -Dorg.gradle.warning.mode=summary distribution
+
+      - name: create snapshots maven folder and copy files
+        run: |
+          mkdir -p maven/org/opensearch/client
```

Comment:
Might much cleaner if you just declare this as a variable/env var and use it around.
Eg: `MAVEN_HOME=maven/org/opensearch/client`
`mkdir -p $MAVEN_HOME`